src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs

```diff
@@ -43,24 +38,38 @@ public static IMvcCoreBuilder AddRazorPages(this IMvcCoreBuilder builder)
             }
 
             builder.AddRazorViewEngine();
+
+            AddFeatureProviders(builder);
             AddServices(builder.Services);
 
             builder.Services.Configure(setupAction);
 
             return builder;
         }
 
+        private static void AddFeatureProviders(IMvcCoreBuilder builder)
+        {
+            if (!builder.PartManager.FeatureProviders.OfType<CompiledPageInfoFeatureProvider>().Any())
+            {
+                builder.PartManager.FeatureProviders.Add(new CompiledPageInfoFeatureProvider());
+            }
+        }
+
         // Internal for testing.
         internal static void AddServices(IServiceCollection services)
         {
             // Options
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<RazorPagesOptions>, RazorPagesOptionsSetup>());
 
-            // Action Invoker
+            // Action description and invocation
             services.TryAddEnumerable(
                 ServiceDescriptor.Singleton<IActionDescriptorProvider, PageActionDescriptorProvider>());
             services.TryAddSingleton<IActionDescriptorChangeProvider, PageActionDescriptorChangeProvider>();
+            services.TryAddEnumerable(
+                ServiceDescriptor.Singleton<IPageMetadataProvider, RazorProjectPageMetadataProvider>());
+            services.TryAddEnumerable(
+                ServiceDescriptor.Singleton<IPageMetadataProvider, CompiledPageMetadataProvider>());
```

Comment:
Take a look at `DefaultApplicationModelProvider` in MVC. This should be similar.